src/utils/IntlWrapper.ts

```diff
@@ -167,119 +141,207 @@ export class IntlWrapper {
-    this.setDate(this.timestamp);
+  // Setters
+  setDate(date: Date | number | string) {
+    this.date = new Date(date);
     return this;
   }
 
+  setLocale(locale: Intl.LocalesArgument) {
+    this.locale = locale;
+    return this;
+  }
+
+  setStyle(options: Intl.DateTimeFormatOptions) {
+    this.dateTimeFormatStyle = options;
+    return this;
+  }
+
+  // Processing
+  protected format() {
+    if (!isValidDate(this.date)) this.text = '';
+    else this.text = this.getDateTimeText(this.date, this.dateTimeFormatStyle);
+    return this;
+  }
+
+  // Intl wrapper functions
+  protected getDateTimeText(
+    date: Date | number,
+    style: Intl.DateTimeFormatOptions | undefined = undefined,
+  ) {
+    return new Intl.DateTimeFormat(this.locale, style).format(date);
+  }
+
+  // Utility
+  isValidDate() {
+    return isValidDate(this.date);
+  }
+
   // Getters
-  Duration = {
-    get: (style: durationFormatStyle = 'narrow'): string => {
-      this.transform(Rules.DurationFormat).format(Rules.DurationFormat, {
-        durationStyle: style,
-      });
-      return this.output.text;
-    },
-  };
+  getDate() {
+    return this.date;
+  }
 
-  Progress = {
-    get: (style: durationFormatStyle = 'narrow') => {
-      this.transform(Rules.Progress).format(Rules.Progress, { durationStyle: style });
-      return { time: this.output.text, isFuture: this.output.progress.isFuture };
-    },
-  };
+  getLocale() {
+    return this.locale;
+  }
 
-  DateTime = {
-    get: (style?: Intl.DateTimeFormatOptions) => {
-      this.transform(Rules.DateTime).format(Rules.DateTime, {
-        dateTimeStyle: style,
-      });
-      return this.output.text;
-    },
-    Time: {
-      get: (style: Intl.DateTimeFormatOptions['timeStyle'] = 'short') => {
-        this.transform(Rules.DateTime).format(Rules.DateTime, {
-          dateTimeStyle: { timeStyle: style },
-        });
-        return this.output.text;
-      },
-    },
-    Date: {
-      get: (style: Intl.DateTimeFormatOptions['dateStyle'] = 'medium') => {
-        this.transform(Rules.DateTime).format(Rules.DateTime, {
-          dateTimeStyle: { dateStyle: style },
-        });
-        return this.output.text;
-      },
-    },
-  };
+  getStyle() {
+    return this.dateTimeFormatStyle;
+  }
 
-  Range = {
-    get: (style?: Intl.DateTimeFormatOptions) => {
-      this.transform(Rules.Range).format(Rules.Range, { dateTimeStyle: style });
-      return this.output.text;
-    },
-    Time: {
-      get: (style: Intl.DateTimeFormatOptions['timeStyle'] = 'short') => {
-        this.transform(Rules.Range).format(Rules.Range, { dateTimeStyle: { timeStyle: style } });
-        return this.output.text;
-      },
-    },
-    Date: {
-      get: (style: Intl.DateTimeFormatOptions['dateStyle'] = 'medium') => {
-        this.transform(Rules.Range).format(Rules.Range, { dateTimeStyle: { dateStyle: style } });
-        return this.output.text;
-      },
-    },
-  };
+  getRelative(
```

Comment:
Ok, lets me again clarify this, summarizing all above what you mean is:
1. `IntlDuration` should be created from `durationFormat` object ONLY
2. If we call it from `IntlDateTime` using Date object we pass it like this - `new IntlDuration({milliseconds: this.date.getTime()})`
3. Dont handle now/future in `IntlDuration` but create 2 functions in `IntlDateTime` 1st could return "Now" and isFuture state and 2nd could not (Maybe just do 1 function but with param?)

Reply:
@DFreezerR 

1 + 2. I see the input of IntlDuration was duration + timestamp. I was thinking it was duration + relative time (because of the setRelativeTime function). But I see the problem you have.
We have 3 types of input.
- {minutes: 150} -> 2h 30min // apply an format
- {hours: 2, minutes: 30} -> 2h 30min // output the duration object like passed
- 9.000.000 -> 2h 30min (3. is not 100% necessary but as it it makes sense to handle it like that internaly why not expose it) // ms relative time

I think in this case it would be the best to remove the input from the constructor and instead have a set function for it. So we have multiple entrypoints.

```ts
(new IntlDuration()).setFormatedDuration({{hours: 1, minutes: 90}, 'M/H/D/Y'); -> 2h 30min
(new IntlDuration()).setDuration({hours: 2, minutes: 30}); -> 2h 30min
(new IntlDuration()).setRelativeTime(9.000.000, 'ms',  'M/H/D/Y' ); -> 2h 30min
```
```ts
  setDurationFormated(duration, style: 'normal' | 'progress' | 'm/h/d/y' = 'normal') {
    const relativeTime = durationToMs(duration);
    this.setRelativeTime(relativeTime, 'ms', style);
    return this;
  }

  setRelativeTime(relativeTime: number, convertFrom?: keyof typeof dateUnitToMs, style: 'normal' | 'progress' | 'm/h/d/y' = 'normal') {
    const duration = this.relativeTimeToDuration(relativeTime, style); // Transforms and applies the format
    this.setDuration(duration);
    return this;
  }

  setDuration(duration) {
    this.duration = duration;
    return this;
  }
```

4.
Yes it should be compared to current date.

3.
Return an boolean for both.

Reply:
> I think in this case it would be the best to remove the input from the constructor and instead have a set function for it.

Yeah, I think I had something like this in the beginning
Oh well, will make adjustments , since I now understood what you tried to say

Hope you don't mind our "slow walking" with this PR